setup.py

```diff
@@ -58,11 +58,13 @@ def main():
                 'futures==3.1.1',
                 'functools32>=3.2,<=3.99'
             ],
-            'pubnub': ['pubnub>=4.0.0,<=4.99']
+            'pubnub': ['pubnub>=4.0.0,<=4.99'],
+            'joblib': ['joblib>=0.11.0,<=0.11.99'],
```

Comment:
Deprecate which behavior? You mean make these both required? That sounds reasonable to me, since they're both pretty light. We've made an effort to keep dependencies to a minimum in the past, and I was sticking with that.